### DISCUSSION TITLE
❓How can i train a model without abnormal data?

### DISCUSSION THREAD

**Discussion by mashuai191** [open-edge-platform/anomalib#1614] [Q&A]
### Describe the bug

i have no abnormal data, but i want to localize or segment the abnormal area of a pic. how can i do that? i tried with below config but seems it doesn't work.

### Dataset

Folder

### Model

PADiM

### Steps to reproduce the behavior

follow the same steps from official github link

### OS information

OS information:
- OS: Win11
- Python version: 3.10
- Anomalib version: [e.g. 0.3.6]
- PyTorch version: [e.g. 1.9.0]
- CUDA/cuDNN version: 12.1
- GPU models and configuration: [e.g. 2x GeForce RTX 3090]
- Any other relevant information: [e.g. I'm using a custom dataset]

### Expected behavior

can do localization of the abnormal area of a pic

### Screenshots

![01](https://github.com/openvinotoolkit/anomalib/assets/40193779/d59ba99c-539b-4443-bea1-0f86eac3e5ab)

### Pip/GitHub

pip

### Configuration YAML

```yaml
dataset:
  name: hazelnut_try
  format: folder
  path: ./datasets/hazelnut_toy
  normal_dir: good # name of the folder containing normal images.
  abnormal_dir: colour # name of the folder containing abnormal images.
  #abnormal_dir: null
  normal_test_dir: null # name of the folder containing normal test images.
  #mask_dir: mask/colour # optional
  mask_dir: null
  task: segmentation # classification or segmentation
  extensions: null
  train_batch_size: 32
  eval_batch_size: 32
  num_workers: 8
  image_size: 256 # dimensions to which images are resized (mandatory)
  center_crop: null # dimensions to which images are center-cropped after resizing (optional)
  normalization: imagenet # data distribution to which the images will be normalized: [none, imagenet]
  transform_config:
    train: null
    eval: null
  #test_split_mode: from_dir # options: [from_dir, synthetic]
  test_split_mode: none
  test_split_ratio: 0.2 # fraction of train images held out testing (usage depends on test_split_mode)
  val_split_mode: same_as_test # options: [same_as_test, from_test, synthetic]
  val_split_ratio: 0.5 # fraction of train/test images held out for validation (usage depends on val_split_mode)
  tiling:
    apply: false
    tile_size: null
    stride: null
    remove_border_count: 0
    use_random_tiling: False
    random_tile_count: 16

model:
  name: padim
  backbone: resnet18
  pre_trained: true
  layers:
    - layer1
    - layer2
    - layer3
  normalization_method: min_max # options: [none, min_max, cdf]

metrics:
  image:
    - F1Score
    - AUROC
  pixel:
    - F1Score
    - AUROC
  threshold:
    image_default: 3
    pixel_default: 3
    adaptive: true

visualization:
  show_images: False # show images on the screen
  save_images: True # save images to the file system
  log_images: True # log images to the available loggers (if any)
  image_save_path: null # path to which images will be saved
  mode: full # options: ["full", "simple"]

project:
  seed: 42
  path: ./results

logging:
  logger: [] # options: [comet, tensorboard, wandb, csv] or combinations.
  log_graph: false # Logs the model graph to respective logger.

optimization:
  export_mode: null # options: torch, onnx, openvino

# PL Trainer Args. Don't add extra parameter here.
trainer:
  enable_checkpointing: true
  default_root_dir: null
  gradient_clip_val: 0
  gradient_clip_algorithm: norm
  num_nodes: 1
  devices: 1
  enable_progress_bar: true
  overfit_batches: 0.0
  track_grad_norm: -1
  check_val_every_n_epoch: 1 # Don't validate before extracting features.
  fast_dev_run: false
  accumulate_grad_batches: 1
  max_epochs: 1
  min_epochs: null
  max_steps: -1
  min_steps: null
  max_time: null
  limit_train_batches: 1.0
  limit_val_batches: 1.0
  limit_test_batches: 1.0
  limit_predict_batches: 1.0
  val_check_interval: 1.0 # Don't validate before extracting features.
  log_every_n_steps: 50
  accelerator: auto # <"cpu", "gpu", "tpu", "ipu", "hpu", "auto">
  strategy: null
  sync_batchnorm: false
  precision: 32
  enable_model_summary: true
  num_sanity_val_steps: 0
  profiler: null
  benchmark: false
  deterministic: false
  reload_dataloaders_every_n_epochs: 0
  auto_lr_find: false
  replace_sampler_ddp: true
  detect_anomaly: false
  auto_scale_batch_size: false
  plugins: null
  move_metrics_to_cpu: false
  multiple_trainloader_mode: max_size_cycle
```

### Logs

```shell
no
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by blaz-r** [comment]
Hello. All models implemented inside Anomalib support training without abnormal data. I am able to train the model with your config, could you share more details what seems to be the problem on your end?

**Comment by mashuai191** [comment]
hi @blaz-r, 
thanks for reply! yes i can train without abnormal data, but i cannot get the localization/segmentation. please see the attached sreenshot.
seems segmentation labels are required for training? if so that is superised training then? 

BR, Shuai Ma

**Comment by blaz-r** [comment]
Hello. No, labels are not required and Padim is an unsupervised model. You should get segmented results inside the results folder. If that is what your screenshot depicts, could you please post the entire image (heatmap, ground truth...).

**Comment by samet-akcay** [comment]
Hi @mashuai191, I see that you commented out the following line in your config

```yaml
#test_split_mode: from_dir # options: [from_dir, synthetic]
```

If you set this to `synthetic`, anomalib would generate some synthetic abnormal images using your normal data. You could then you the `segmentation` task. Hope this helps!

**Comment by mashuai191** [comment]
thanks @samet-akcay @blaz-r for your help! i will try at the weekend.
